### PR TITLE
YAML test framework: simplify version node_selector

### PR DESCRIPTION
**Pull request by ldematte**
Remove ranges and allow just `current`/`original`.

Node selectors with version ranges are used today just for mixed cluster/rolling upgrade tests, to filter out (or select) the new upgraded nodes that exhibit a change behaviour.
Therefore it is not needed to specify explicitly a version range: selecting nodes that have been upgraded (`current`) or not (`original`) is sufficient.

Today just `current` exists; this PR adds `original` (with one example usage) and removes the ability to select nodes based on version.

(Follows https://github.com/elastic/elasticsearch/pull/103889, where we removed all version-based `node_selector`s but this one)

### CODE REVIEW

**Comment by mosche** [comment]
This makes sense to me @ldematte 👍  But please gather some more opinions :)
As a suggestion, how about using `previous` instead of the negated `non_current`?

**Comment by ldematte** [comment]
Naming is hard :)
In code, we current have "original" and "old" to describe the "not current" cluster. "Previous" can be confused with some `-1` version (previous minor like 8.12 or previous major like 7). Maybe if we want to avoid the negation, `original` is a better choice.

**Comment by ldematte** [comment]
> As a suggestion, how about using previous instead of the negated non_current?

I will change that to `original` after merging https://github.com/elastic/elasticsearch/pull/103889 - it makes more sense in the context of upgraded/mixed clusters. Wdyt @mosche?

**Comment by mosche** [comment]
Is this only for mixed cluster tests, @ldematte? Initially I was struggling to bridge the gap between `original` and `current`, but the more I think about it the better I like it 👍  
Even better would be something like `original` (or also `initial`) and `upgraded` instead of `current`. That way both terms are not about a specific version, but more about the state in terms of upgrading.

**Comment by ldematte** [comment]
AFAIK, it has a meaning only in mixed cluster and rolling upgrade tests. I like the change to `upgraded` too - will have to be careful with that/do it in 2 phases as I think it is used in (and was introduced for) serverless.

File: test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/DoSectionTests.java [hunk 1]
```diff
@@ -579,82 +571,48 @@ public void testParseDoSectionAllowedWarnings() throws Exception {
...
+            endsWith("unknown version selector [5.2.0-6.0.0]. Only [current] and [non_current] are allowed.")
         );
-
-        {
-            List<Node> badNodes = new ArrayList<>();
-            badNodes.add(new Node(new HttpHost("dummy")));
-            Exception e = expectThrows(IllegalStateException.class, () -> doSection.getApiCallSection().getNodeSelector().select(badNodes));
-            assertEquals("expected [version] metadata to be set but got [host=http://dummy]", e.getMessage());
-        }
+        parser.close();
```
Comment (williamrandolph) [inline test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/DoSectionTests.java:586]: Do we want to let the `AbstractClientYamlTestFragmentParserTestCase` do its check here?

**Review by williamrandolph — COMMENTED** [review]
Just one question, but LGTM otherwise.

File: test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/DoSectionTests.java (hunk 1 above)
Comment (ldematte) [reply test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/DoSectionTests.java]: Yes, I changed it and caught a pre-existing problem. Nice!

**Comment by ldematte** [comment]
I had to re-introduce the version matching for restCompat tests. Those test are a bit peculiar: they checkout `elastic/7.17` to get the yaml files/specs from that branch, but run `elastic/main` test code.
So I added a check to see if we are in a restCompat tests, and only if we are, we do go down the legacy parser that support versions. This would prevent new usages of version-based selectors to sneak in, and also made removal once we switch to 9 (I have decorated the relevant methods with `@UpdateForV9`)

**Comment by ldematte** [comment]
Serverless tests fail for an unrelated, known issue: https://github.com/elastic/elasticsearch-serverless/issues/1461

**Review by williamrandolph — APPROVED** [review]
LGTM!

**Pull request merged**